api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -116,34 +124,48 @@ type FelixConfigurationSpec struct {
 	// processes that also take the lock. When running Felix inside a container, this requires the
 	// /run directory of the host to be mounted into the calico/node or calico/felix container.
 	// [Default: 0s disabled]
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
 	IptablesLockTimeout *metav1.Duration `json:"iptablesLockTimeout,omitempty" configv1timescale:"seconds" confignamev1:"IptablesLockTimeoutSecs"`
 	// IptablesLockProbeInterval is the time that Felix will wait between
 	// attempts to acquire the iptables lock if it is not available. Lower values make Felix more
 	// responsive when the lock is contended, but use more CPU. [Default: 50ms]
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
 	IptablesLockProbeInterval *metav1.Duration `json:"iptablesLockProbeInterval,omitempty" configv1timescale:"milliseconds" confignamev1:"IptablesLockProbeIntervalMillis"`
 	// FeatureDetectOverride is used to override feature detection based on auto-detected platform
 	// capabilities.  Values are specified in a comma separated list with no spaces, example;
 	// "SNATFullyRandom=true,MASQFullyRandom=false,RestoreSupportsLock=".  "true" or "false" will
 	// force the feature, empty or omitted values are auto-detected.
+	// +kubebuilder:validation:Pattern="^([a-zA-Z]+=(true|false|),)*([a-zA-Z]+=(true|false|))$"
 	FeatureDetectOverride string `json:"featureDetectOverride,omitempty" validate:"omitempty,keyValueList"`
 	// FeatureGates is used to enable or disable tech-preview Calico features.
 	// Values are specified in a comma separated list with no spaces, example;
 	// "BPFConnectTimeLoadBalancingWorkaround=enabled,XyZ=false". This is
 	// used to enable features that are not fully production ready.
+	// +kubebuilder:validation:Pattern="^([a-zA-Z]+=(enabled|disabled|true|false),)*([a-zA-Z]+=(enabled|disabled|true|false))$"
```

Comment:
the list of values here is tricky, we have a gate atm with `udp` as a value. Perhaps anything that does not include `=` ? idk :thinking:

Reply:
replacing the regex with ^([a-zA-Z]+=(enabled|disabled|true|false),)*([a-zA-Z]+=([^=]+))$ to allow any character other than "=" for values.